src/Backups/BackupIO_S3.cpp

```diff
@@ -60,6 +64,7 @@ namespace
         client_configuration.connectTimeoutMs = 10 * 1000;
         /// Requests in backups can be extremely long, set to one hour
         client_configuration.requestTimeoutMs = 60 * 60 * 1000;
+        client_configuration.retryStrategy = std::make_shared<Aws::Client::DefaultRetryStrategy>(request_settings.retry_attempts);
```

Comment:
That is disaster.
That strategy doesn't have a limitation.
see https://github.com/ClickHouse/ClickHouse/pull/54651
and that https://github.com/ClickHouse/clickhouse-private/issues/3925

Reply:
Ok, I'll replace `Aws::Client::RetryStrategy` with our `S3::Client::RetryStrategy`

Reply:
I'm wrong here. Sorry for the fuz.
ClientFactory::create overrides the `retryStrategy`.